src/cluster_asm.c

```diff
@@ -1012,3 +1010,64 @@ int slotRangesSnapshotSaveRio(int req, rio *rdb, int *error) {
     if (error) *error = errno;
     return C_ERR;
 }
+
+/* ======================== ASM Sync Buffer Functions ======================== */
+
+/* Read error handler for sync buffer */
+static void asmReadSyncBufferErrorHandler(connection *conn) {
+    serverLog(LL_WARNING, "ASM buffer stream: error while reading from source: %s",
+              connGetLastError(conn));
+    asmTask *task = connGetPrivateData(conn);
+
+    task->state = ASM_FAILED;
+    replDataBufFree(&task->sync_buffer);
+    if (task->rdb_channel_conn) connClose(task->rdb_channel_conn);
+    if (task->main_channel_conn) connClose(task->main_channel_conn);
+}
+
+/* Read data from connection into sync buffer. */
+static void asmSyncBufferReadFromConn(connection *conn) {
+    asmTask *task = connGetPrivateData(conn);
+    if (!task) {
+        serverLog(LL_WARNING, "ASM buffer stream: no task associated with connection");
+        return;
+    }
+
+    replDataBufReadFromConn(conn, &task->sync_buffer, asmReadSyncBufferErrorHandler);
+}
+
+static void asmSyncBufferStreamYieldCallback(void *ctx) {
+    replDataBufToDbCtx *context = (replDataBufToDbCtx *) ctx;
```

Comment:
```suggestion
    replDataBufToDbCtx *context = ctx;
```

nitpick :upside_down_face:

Reply:
OK